en/docs/develop-components/develop-services/develop-a-grpc-service.md

```diff
@@ -0,0 +1,210 @@
+
+         - Read and write access to code and pull requests.
+         - Read access to issues and metadata.
+             
+          You can [revoke access](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/reviewing-your-authorized-integrations#reviewing-your-authorized-github-apps) if you do not want Choreo to have access to your GitHub account. However, write access is only used to send pull requests to a user repository. Choreo will not directly push any changes to a repository.
+
+8. In the **Connect Repository** pane, enter the following information:
+
+    | **Field**               | **Description**                 |
+    |-------------------------|---------------------------------|
+    | **GitHub Account**      | Your account                    |
+    | **GitHub Repository**   | **`choreo-sample-apps`**        |
+    | **Branch**              | **`main`**                      |
+    | **Build Preset**        | **Dockerfile**                  |
+    | **Dockerfile path**     | **go/grpc/Dockerfile.server** |
+    | **Docker context path** | **go/grpc**                     |
+
+
+    !!! info
+        1.  To successfully build your container with Choreo, it is essential to explicitly define a User ID (UID) under the USER instruction in your Dockerfile. For reference, see [sample Dockerfile](https://github.com/wso2/choreo-sample-apps/blob/main/go/greeter/Dockerfile).
+        To ensure that the defined USER instruction is valid, it must conform to the following conditions:
+            - A valid User ID is a numeric value between 10000-20000, such as `10001` or `10500`.
+            - Usernames are considered invalid and should not be used. For example, `my-custom-user-12221` or `my-custom-user` are invalid User IDs.
+
+        2. The Dockerfile utilized in this guide is a Multi-stage Dockerfile, which is designed to keep the final image size small and provides the ability to build the application with a specific version of tools and libraries.
+
+9. Click **Create**. Once the component creation is complete, you will see the component's overview page.
+
+You have successfully created a Service component from a Dockerfile. Now let's build and deploy the service.
+
+## Step 2: Configure the service port with endpoints
+
+We expect to run our gRPC server Service component on port 8080. To securely expose the service through Choreo, we must provide the port and other required information to Choreo. In Choreo, we expose our services with endpoints. You can read more about endpoints in our [endpoint documentation](https://wso2.com/choreo/docs/develop-components/develop-services/develop-a-service/#what-are-endpoints-in-service-components).
+
+Choreo looks for an endpoints.yaml file inside the `.choreo` directory to configure the endpoint details of a containerized component. Place the `.choreo` directory at the root of the Docker build context path.
```

Comment:
endpoints.yaml should be in code format.

Reply:
done